search/indexing_api.py

```diff
@@ -370,34 +369,60 @@ def recreate_index():
     """
     Wipe and recreate index and mapping, and index all items.
     """
-    temp_index = get_temp_index()
-    clear_index(temp_index)
+    conn = get_conn(verify=False)
+
+    # Create new backing index for reindex
+    new_backing_index = "{index}_{uuid}".format(
+        index=settings.ELASTICSEARCH_INDEX,
+        uuid=uuid4().hex,
+    )
+    clear_index(new_backing_index)
+
+    # Clear away temp alias so we can reuse it
+    temp_alias = get_temp_alias()
+    if conn.indices.exists_alias(name=temp_alias):
+        # Deletes both alias and backing indexes
+        conn.indices.delete(temp_alias)
```

Comment:
This looks like a call to delete an index but an alias name is being passed here, so I don't think it actually would do anything in practice.